### PR TITLE
Move fixture-properties:clean function into default overrides

### PR DESCRIPTION
**Pull request by frenchy64**
Related https://github.com/threatgrid/iroh/issues/3986

Deletes `fixture-properties:clean` and moves its overrides to be the default overrides for all tests. This removes some global spaghetti logic used to create test configs, and is a useful step towards more localised construction (rather than dynamic bindings).

<a name="qa">[§](#qa)</a> QA
============================

No QA is needed.

<a name="release-notes">[§](#release-notes)</a> Release Notes
=============================================================

```
intern: Move fixture-properties:clean function into default overrides
```

<a name="squashed-commits">[§](#squashed-commits)</a> Squashed Commits
======================================================================

### CODE REVIEW

File: test/ctia/entity/web_test.clj [hunk 1]
```diff
@@ -261,12 +260,13 @@
...
 (s/defn apply-fixtures-with-app
-  [properties f-with-app :-  (s/=> s/Any (s/=> s/Any (s/named s/Any 'app)))]
+  [properties f-with-app :- (s/=> s/Any (s/=> s/Any (s/named s/Any 'app)))]
   (let [fixture-fn
         (join-fixtures [helpers/fixture-log
-                        helpers/fixture-properties:clean
                         helpers/fixture-properties:cors
-                        #(helpers/with-properties properties (%))
+                        #(helpers/with-properties
+                           (conj properties "ctia.auth.type" "allow-all")
```
Comment (frenchy64) [inline test/ctia/entity/web_test.clj:268]: There were 2 overlapping calls to `helpers/fixture-properties:clean` in this ns that this PR removes. AFAICT `ctia.auth.type` is the only important property for the latter call, so a straightforward fix is to just set it explicitly.

File: test/ctia/task/rollover_test.clj [hunk 2]
```diff
@@ -12,11 +12,12 @@
...
   (join-fixtures [whoami-helpers/fixture-server
                   whoami-helpers/fixture-reset-state
-                  helpers/fixture-properties:clean
                   es-helpers/fixture-properties:es-store]))
 
 (use-fixtures :each
-  (join-fixtures [helpers/fixture-ctia
+  (join-fixtures [#(helpers/with-properties
+                     ["ctia.auth.type" "allow-all"]
+                     (helpers/fixture-ctia %))
```
Comment (frenchy64) [inline test/ctia/task/rollover_test.clj:20]: Override `whoami-helpers/fixture-server` wrapper in `:once` fixture.

File: test/ctia/task/migration/store_test.clj [hunk 3]
```diff
@@ -268,7 +267,9 @@
           (ductile.index/delete! (str (migration-index get-in-config) "*")))))))
 
 (use-fixtures :each
-  (join-fixtures [helpers/fixture-ctia
+  (join-fixtures [#(helpers/with-properties
+                     ["ctia.auth.type" "allow-all"]
+                     (helpers/fixture-ctia %))
```
Comment (frenchy64) [inline test/ctia/task/migration/store_test.clj:272]: Override `whoami-helpers/fixture-server` wrapper in `:once` fixture.

**Comment by frenchy64** [comment]
I noticed the tests are actually slower with the `min-threads` change. I'll pull that out.

**Review by vemv — APPROVED** [review]
LGTM!

**Comment by frenchy64** [comment]
Thanks Victor!

**Pull request merged**